From oliver@neukum.org Thu Dec 20 03:22:46 2007
From: Oliver Neukum <oliver@neukum.org>
Date: Thu, 20 Dec 2007 12:23:03 +0100
Subject: USB: fix NULL pointer dereference on drivers/usb/serial/whiteheat.c
To: linux-usb-devel@lists.sourceforge.net, Greg KH <greg@kroah.com>
Cc: Andrew Morton <akpm@linux-foundation.org>, marciobuss@gmail.com, bugme-daemon@bugzilla.kernel.org
Message-ID: <200712201223.04307.oliver@neukum.org>
Content-Disposition: inline


Am Sonntag, 16. Dezember 2007 05:23:47 schrieb Andrew Morton:
> On Sat, 15 Dec 2007 07:08:52 -0800 (PST) bugme-daemon@bugzilla.kernel.org wrote:
> 
> > http://bugzilla.kernel.org/show_bug.cgi?id=9573
 
> > There's a null pointer dereference on drivers/usb/serial/whiteheat.c as
> > follows:
> > 
> > (1) line 613:  the test "if (port->tty)" implies that null is a legal value
> >     for "port->tty" at that point
> > 
> > (2) neither firm_open nor firm_purge initialize "port->tty"

drivers/usb/serial/usb-serial::serial_open() sets port->tty
The check for NULL is bogus. This patch removes the check.

Signed-off-by: Oliver Neukum <oneukum@suse.de>
Acked-by: Alan Cox <alan@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/serial/whiteheat.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/usb/serial/whiteheat.c
+++ b/drivers/usb/serial/whiteheat.c
@@ -610,8 +610,7 @@ static int whiteheat_open (struct usb_se
 	if (retval)
 		goto exit;
 
-	if (port->tty)
-		port->tty->low_latency = 1;
+	port->tty->low_latency = 1;
 
 	/* send an open port command */
 	retval = firm_open(port);
